src/Trace/Tracer/ExtensionTracer.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace\Tracer;
+
+use Google\Cloud\Core\ArrayTrait;
+use Google\Cloud\Trace\TraceContext;
+use Google\Cloud\Trace\TraceSpan;
+
+/**
+ * This implementation of the TracerInterface utilizes the stackdriver extension
```

Comment:
Apologies if I missed this somewhere:

A little detail as to which environments this extension can be expected to be available in would be really helpful :)